libs/ledger-live-common/src/families/near/banner.ts

```diff
@@ -0,0 +1,85 @@
+import { getCurrentNearPreloadData } from "./preload";
+import { canStake, canUnstake, FIGMENT_NEAR_VALIDATOR_ADDRESS } from "./logic";
+import type { NearAccount, NearValidatorItem } from "./types";
+
+export interface AccountBannerState {
+  display: boolean;
+  redelegate: boolean;
+  validatorId: string;
+  ledgerValidator: NearValidatorItem | undefined;
+}
+
+export function getAccountBannerState(
+  account: NearAccount
+): AccountBannerState {
+  // Group current validator
+  const nearResources = account.nearResources
+    ? account.nearResources
+    : { stakingPositions: [] };
+
+  const delegations = nearResources?.stakingPositions.map((delegation) => {
+    return delegation;
+  });
+
+  // Get ledger validator data
+  const { validators } = getCurrentNearPreloadData() ?? {
+    validators: [],
+  };
+
+  const ledgerValidator = validators.find(
+    (validator) => validator.validatorAddress === FIGMENT_NEAR_VALIDATOR_ADDRESS
+  );
+
+  // If Ledger doesn't provide validator, we don't display banner
```

Comment:
Do we need to hide the banner if `ledgerValidator` is not found? I can see that we're doing this for the other banners, but I don't see this in the acceptance criteria for these tickets 🤔

Reply:
Not really, but in case that ledger validator address change (without change common variable) or close, we are covered. Theoretically, this is not going to happen.